Sources/Utility/FSWatch.swift

```diff
@@ -0,0 +1,602 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Basic
+import Dispatch
+import Foundation
+import SPMLibc
+
+public protocol FSWatchDelegate {
+    // FIXME: We need to provide richer information about the events.
+    func pathsDidReceiveEvent(_ paths: [AbsolutePath])
```

Comment:
Shouldn't there also be some indication what event was associated with the path, e.g. 'created', 'deleted', etc. ?

Reply:
Yeah, thats what the above FIXME is for. We need to enhance the API send more information about events. That can be done incrementally.